modules/ROOT/pages/_partials/sync/server-deploying-your-app-built-using-the-data-sync-framework-to-a-development-environment.adoc

```diff
@@ -0,0 +1,275 @@
+For more information, see xref:server-getting-started-with-an-example-data-source-{context}[Getting started with an example data source].
+
+.Prerequisites
+
+* You have a running environment.
+* You have cloned the link:https://github.com/aerogear/voyager-server[Apollo Voyager Server] locally.
+* An understanding of `git`.
+
+.Procedure
+
+. If the {sync-server} is running, stop it.
++
+. Copy the contents from `index.js` and use this code to replace the contents in `examples/basic/server.js`.
++
+. Save these changes and commit the changes to a github repository.
++
+. Create a blank `yaml` file locally.
++
+. Open the `yaml` file and copy the following code into the `yaml` file:
++
+[source,yaml]
+----
+
+---
+kind: Template
+apiVersion: v1
+metadata:
+  name: datasync-server-app
+  annotations:
+    openshift.io/display-name: Data Sync App
+    description: |-
+          This template allows for the deployment of the DataSync Server App from source code.
+    iconClass: icon-nodejs
+    openshift.io/provider-display-name: Red Hat, Inc.
+    openshift.io/documentation-url: https://docs.aerogear.org/aerogear/latest/data-sync.html#sync-server-openshift
+    openshift.io/support-url: https://access.redhat.com
+    template.openshift.io/bindable: 'false'
+message: 'The following service(s) have been created in your project: ${NAME}.'
+labels:
+  template: data-sync-server-app
+  app: data-sync-server-app
+objects:
+- kind: Service
+  apiVersion: v1
+  metadata:
+    name: "${NAME}"
+    annotations:
+      description: Exposes and load balances the application pods
+  spec:
+    ports:
+    - name: http
+      port:  ${{PORT}}
+      targetPort:  ${{PORT}}
+    selector:
+      name: "${NAME}"
+- kind: Route
+  apiVersion: v1
+  metadata:
+    name: "${NAME}"
+  spec:
+    host: "${APPLICATION_DOMAIN}"
+    to:
+      kind: Service
+      name: "${NAME}"
+- kind: ImageStream
+  apiVersion: v1
+  metadata:
+    name: "${NAME}"
+    annotations:
+      description: Keeps track of changes in the application image
+- kind: BuildConfig
+  apiVersion: v1
+  metadata:
+    name: "${NAME}"
+    annotations:
+      description: Defines how to build the application
+      template.alpha.openshift.io/wait-for-ready: 'true'
+  spec:
+    source:
+      type: Git
+      git:
+        uri: "${SOURCE_REPOSITORY_URL}"
+        ref: "${SOURCE_REPOSITORY_REF}"
+      contextDir: "${CONTEXT_DIR}"
+    strategy:
+      sourceStrategy:
+        from:
+          kind: ImageStreamTag
+          name: "${BUILDER_IMAGE}"
+          namespace: openshift
+        env:
+        - name: NPM_MIRROR
+          value: "${NPM_MIRROR}"
+    output:
+      to:
+        kind: ImageStreamTag
+        name: "${NAME}:latest"
+    triggers:
+    - type: ImageChange
+    - type: ConfigChange
+    - type: GitHub
+      github:
+        secret: "${GITHUB_WEBHOOK_SECRET}"
+    - type: Generic
+      generic:
+        secret: "${GENERIC_WEBHOOK_SECRET}"
```

Comment:
It looks like we didn't actually use these triggers in this doc, I don't think we should keep them here. If we want to keep them, then we probably should add a link in this doc to explain how to use the webhooks: https://docs.openshift.com/container-platform/3.9/dev_guide/builds/triggering_builds.html#github-webhooks

Reply:
The snippet of code was updated, and successfully tested: https://github.com/aerogear/mobile-docs/pull/694/commits/f2dbb86cfc89fda90db3f1b7fc42293b2b23b98c